clash-lib/src/Clash/Normalize/Transformations.hs

```diff
@@ -2558,23 +2577,29 @@ inlineBndrsCleanup isN origInl = go
   go doneInl [] =
     -- If some of the let-binders that we wanted to inline turn out to be
     -- recursive, then we have to keep those around as well, as we weren't able
-    -- to inline them.
-    [ (v,e) | ((v,e),_,Rec) <- eltsVarEnv doneInl ]
-  go !doneInl (((v,e),eFVs):il) =
-    let (sM,_,doneInl1) = foldlWithUniqueVarEnv'
+    -- to inline them. Furthermore, for every recursive binder there might still
+    -- be non-inlined variables left, see #1337.
+    flip map [ (ve, eFvs) | (ve,eFvs,Rec) <- eltsVarEnv doneInl ] $ \((v, e), eFvs) ->
+      let
+        (substM, _, _) = foldlWithUniqueVarEnv'
+                           (reduceBindersCleanup isN origInl)
```

Comment:
Shouldn't we pass in en empty varenv instead of `origInl`? I mean... at this point we don't want to lookup things in `origInl` anymore.

Reply:
True, that should be fine!